app/controllers/WhatYouOweController.scala

```diff
@@ -59,65 +56,15 @@ class WhatYouOweController @Inject()(val authActions: AuthActions,
-          itvcErrorHandler.showInternalServerError()
-        case (Right(startUrl), Some(lpp2Url)) =>
-
-          val wyoViewModel: WhatYouOweViewModel = WhatYouOweViewModel(
-            currentDate = dateService.getCurrentDate,
-            hasOverdueOrAccruingInterestCharges = hasOverdueCharges || hasAccruingInterestReviewAndReconcileCharges,
-            whatYouOweChargesList = whatYouOweChargesList,
-            hasLpiWithDunningLock = whatYouOweChargesList.hasLpiWithDunningLock,
-            currentTaxYear = dateService.getCurrentTaxYearEnd,
-            backUrl = backUrl,
-            utr = user.saUtr,
-            dunningLock = whatYouOweChargesList.hasDunningLock,
-            creditAndRefundUrl = (user.isAgent() match {
-              case true if user.incomeSources.yearOfMigration.isDefined  => CreditAndRefundController.showAgent()
-              case true                                                  => NotMigratedUserController.showAgent()
-              case false if user.incomeSources.yearOfMigration.isDefined => CreditAndRefundController.show()
-              case false                                                 => NotMigratedUserController.show()
-            }).url,
-            creditAndRefundEnabled = isEnabled(CreditsRefundsRepay),
-            taxYearSummaryUrl = {
-              if (user.isAgent()) TaxYearSummaryController.renderAgentTaxYearSummaryPage(_).url
-              else                TaxYearSummaryController.renderTaxYearSummaryPage(_, origin).url
-            },
-            claimToAdjustViewModel = ctaViewModel,
-            lpp2Url = lpp2Url,
-            adjustPoaUrl = controllers.claimToAdjustPoa.routes.AmendablePoaController.show(user.isAgent()).url,
-            chargeSummaryUrl = (taxYearEnd: Int, transactionId: String, isInterest: Boolean, origin: Option[String]) => {
-              if (user.isAgent()) ChargeSummaryController.showAgent(taxYearEnd, transactionId, isInterest).url
-              else                ChargeSummaryController.show(taxYearEnd, transactionId, isInterest, origin).url
-            },
-            paymentHandOffUrl = PaymentController.paymentHandoff(_, origin).url,
-            selfServeTimeToPayEnabled  = isEnabled(SelfServeTimeToPayR17),
-            selfServeTimeToPayStartUrl = startUrl
-          )
-          Ok(whatYouOwe(
-            viewModel = wyoViewModel,
-            origin = origin)(user, user, messages, dateService))
-            .addingToSession(gatewayPage -> WhatYouOwePage.name)
-        case (_, None) =>
-          Logger("application").error("No chargeReference supplied with second late payment penalty. Hand-off url could not be formulated")
-          itvcErrorHandler.showInternalServerError()
-      }
+    whatYouOweService.createWhatYouOweViewModel(backUrl, origin, getCreditAndRefundUrl, getTaxYearSummaryUrl(origin), getAdjustPoaUrl, getChargeSummaryUrl, getPaymentHandOffUrl(origin)) map {
```

Comment:
Can remove the `claimToAdjustService`, `selfServeTimeToPayService` and `auditingService` parameters we're injecting into the class as they've been moved to the service layer and therefore not being used here

<img width="792" height="219" alt="Screenshot 2025-08-29 at 10 26 10" src="https://github.com/user-attachments/assets/1f9ba2fc-aa0e-4661-a3d2-afba2d0e3781" />